### ISSUE TITLE
Remote debug over adb seems to be broken (2.1.4-beta)

### ISSUE THREAD

**Issue by DrMoriarty** [godotengine/godot#10458]
**Operating system or device - Godot version:**
OSX, Android 4.4.2
Godot 2.1.4-beta (commit e5102a9)
**Issue description:**

If I set `Use remote debug over Adb` the debugging doesn't work at all.

[There I described the problem](https://godotengine.org/qa/17399/using-remote-debug-on-android-2-1-4-beta)

**Steps to reproduce:**

Set `Use remote debug over Adb` and try to debug on the android device.

**Link to minimal example project:**

**Comment by HummusSamurai** [comment]
Which Godot version are you using? (Public alpha, master branch, etc..)

**Comment by DrMoriarty** [comment]
@HummusSamurai 2.1.4-beta (commit e5102a9)

**Comment by RandomShaper** [comment]
Can you try enabling/disabling the _Force system user_ flag under Android debug settings?

**Comment by RandomShaper** [comment]
Never mind. I've read your problem in Q&A. I'l have a look.

**Comment by eon-s** [comment]
I have had that problem for months (after updating all my systems) on linux and windows, but I was unable to make older versions of Godot to work again either.
Turning off `Use remote debug over Adb` solved the problem for me too.

With the option _on_ just got after "Installing into device":
```
Success
error: closed
error: closed
error: closed
error: closed
Reverse result: 1
error: closed
error: closed
Reverse result2: 1
Starting: Intent { act=android.intent.action.MAIN cmp=org.godotengine.testprj/org.godotengine.godot.Godot }
```

**Comment by eon-s** [comment]
@RandomShaper with your PR I'm not able to use remote debug anymore 

got, after installing:
```
Success
error: closed
error: closed
Reverse result: 1
Starting: Intent { act=android.intent.action.MAIN cmp=org.godotengine.platformer/org.godotengine.godot.Godot }

```

If I don't enable the remote debug, the message is:
```
Success
error: closed
Starting: Intent { act=android.intent.action.MAIN cmp=org.godotengine.platformer/org.godotengine.godot.Godot }
```

**Comment by RandomShaper** [comment]
I pushed again to fix that. Has you fetchef the last revision?

I will be out of the city till Sunday so I won't be able to fix it until then, in case it definitely turns out to be failing.

In the meantime I'll tag it as 'needs testing'.

**Comment by eon-s** [comment]
@RandomShaper I'll check again directly with your repo to avoid my classic mistakes, I will try to test on windows too during the weekend.

**Comment by eon-s** [comment]
Tested the RC1 on windows and has the same symptom and "fix" (previous tests were made on Ubuntu), I'll try to build the PR and see what happens.

**Comment by eon-s** [comment]
Ok, tested on Windows, the 2.1.4 RC1 works with the remote debug _editor option_ disabled.

With the same editor settings #10530 do not work (output is the same on windows and linux, like when the option is enabled on the other).

**Comment by hpvb** [comment]
Can you try again with rc2? I believe some more fixes for this made it there

**Comment by akien-mga** [comment]
There's no rc2 :)

**Comment by hpvb** [comment]
Well, then... I thought that's what you released the other day. I'll just go and euh, wash my hair or something.

**Comment by DrMoriarty** [comment]
@RandomShaper 
The PR https://github.com/godotengine/godot/pull/10530 wich have been merged today makes things even worse.

The debugging on android doesn't work at all and it doesn't matter if I checked "Use remote debug over Adb" or not.

I checked the current 2.1 branch ( commit 213ca00 )

**Comment by eon-s** [comment]
@DrMoriarty If the merged one was the same I have tested a couple of days ago, then yeah, is worse...

**Comment by akien-mga** [comment]
> If the merged one was the same I have tested a couple of days ago, then yeah, is worse...

That was my evil plan to make sure @RandomShaper fixes the issue once and for all, now that he's responsible for worsening it :P

Note that I merged *after* making the 2.1.4-stable binaries, so the previous workaround should still work.

**Comment by RandomShaper** [comment]
I'm taking care of it right now.

After my PR, the 'debug over ADB' setting is no longer part of the engine. It can still appear in the config file (therefore in the settings dialog) if you ever tweaked it, but you can safely remove it.

**Comment by RandomShaper** [comment]
@DrMoriarty, @eon-s, can you show me your logcats when you try to debug?

**Comment by RandomShaper** [comment]
Forgot to add (for e-mail followers):
- Project run by one-click-deploy (hitting the Android icon in the toolbar).

**Comment by RandomShaper** [comment]
Is works for me every time. Remote inspector, sync scene changes, small deploy, etc. All working. I need more info.

Even breakpoints work. That was added in the demonized PR. Not yet as well as they should, but better than before, when they weren't working at all.

**Comment by eon-s** [comment]
@RandomShaper you want to see the logcat when reverse is rejected/display closed?

I never reported this because stopped working suddenly for me (even on older Godot versions) after some system updates/reinstalls, then saw somewhere that adb got changes and that reverse only worked for android 5+ and I have thought that was the case.
The only device I have at hand now are 4.2.2 and 4.4.4.

I always turned on that editor setting, now saw that turning off let me do everything like before.

**Comment by RandomShaper** [comment]
If the issue here is reverse needs 5.0+, no need for logcat.

Anyway I won't put the setting back. Instead Godot will pick one or the other depending on the Android version of the device, with a message about it.

**Comment by eon-s** [comment]
I don' t know if this helps, windows throws some extra lines on the failed reverse, this is using the RC with the (now removed) editor option enabled, the output after install success:

```
running cmdline: "C:/Android/sdk/platform-tools/adb.exe" "-s" "AIJBBMHEUWE64PHQ" "reverse" "--remove-all"
error: closed
running cmdline: "C:/Android/sdk/platform-tools/adb.exe" "-s" "AIJBBMHEUWE64PHQ" "reverse" "tcp:6096" "tcp:6096"
error: closed
Reverse result: 0
running cmdline: "C:/Android/sdk/platform-tools/adb.exe" "-s" "AIJBBMHEUWE64PHQ" "reverse" "tcp:6010" "tcp:6010"
error: closed
Reverse result2: 0
running cmdline: "C:/Android/sdk/platform-tools/adb.exe" "-s" "AIJBBMHEUWE64PHQ" "shell" "am" "start" "-a" "android.intent.action.MAIN" "-n" "org.godotengine.platformer/org.godotengine.godot.Godot"
Starting: Intent { act=android.intent.action.MAIN cmp=org.godotengine.platformer/org.godotengine.godot.Godot }
```

With the option disabled, it just display the last 2 lines (running cmdline and starting).

**Comment by DrMoriarty** [comment]
@RandomShaper 
1) checked out commit 01ef7a73de46236c4b0cd0e7853f9d3c91768111 (current 2.1 HEAD)
2) build godot engine and android templates:

```
scons platform=osx target=release_debug tools=yes bits=64 -j12
export ANDROID_NDK_ROOT=/opt/android-ndk-r14b/
export ANDROID_HOME=/opt/android-sdk/
scons platform=android target=release android_stl=yes -j12
cd platform/android/java
./gradlew build
cd ../../../
scons platform=android target=release_debug android_stl=yes verbose=yes -j12 
cd platform/android/java
./gradlew build
cd ../../../
```
3) copy templates to ~/.godot/templates so engine can found them by default

4) adb version
```
bash-3.2$ adb version
Android Debug Bridge version 1.0.39
Revision 5943271ace17-android
```

5) device is android 4.4.2 (I will be able to check on 5+ or 6+ later)

6) Editor - Android - Force System User disabled

Deploy - Remote debug enabled

Deploy - Small deploy with Network FS enabled

7) Start project with green android button

![2017-08-30 16 18 54](https://user-images.githubusercontent.com/1177068/29874303-fe7e0676-8d9e-11e7-9e26-bd39560c9dd4.png)

8) catlog:

```
08-30 16:20:35.612: I/godot(5539): Remote Debugger: Connection failed with status: '3', retrying in 1 sec.
08-30 16:20:36.612: I/godot(5539): Remote Debugger: Connection failed with status: '3', retrying in 1 sec.
08-30 16:20:37.612: I/godot(5539): Remote Debugger: Connection failed with status: '3', retrying in 1 sec.
08-30 16:20:37.612: I/godot(5539): Remote Debugger: Unable to connect
08-30 16:20:37.616: E/godot(5539): Could not connect to remotefs: localhost:6010
08-30 16:20:37.616: I/godot(5539): Godot Engine v2.1.4.stable.custom_build (c) 2008-2017 Juan Linietsky, Ariel Manzur.
08-30 16:20:37.616: I/godot(5539): Usage: apk [options] [scene]
08-30 16:20:37.616: I/godot(5539): Options:
08-30 16:20:37.616: I/godot(5539): 	-path [dir] : Path to a game, containing engine.cfg
08-30 16:20:37.616: I/godot(5539): 	-test [test] : Run a test.
08-30 16:20:37.616: I/godot(5539): 		(
08-30 16:20:37.616: I/godot(5539): string
08-30 16:20:37.616: I/godot(5539): , containers
08-30 16:20:37.616: I/godot(5539): , math
08-30 16:20:37.616: I/godot(5539): , render
08-30 16:20:37.616: I/godot(5539): , particles
08-30 16:20:37.616: I/godot(5539): , multimesh
08-30 16:20:37.616: I/godot(5539): , gui
08-30 16:20:37.616: I/godot(5539): , io
08-30 16:20:37.616: I/godot(5539): , shaderlang
08-30 16:20:37.616: I/godot(5539): , physics
08-30 16:20:37.616: I/godot(5539): )
08-30 16:20:37.616: I/godot(5539): 	-r WIDTHxHEIGHT	 : Request Window Resolution
08-30 16:20:37.616: I/godot(5539): 	-p XxY	 : Request Window Position
08-30 16:20:37.616: I/godot(5539): 	-f		 : Request Fullscreen
08-30 16:20:37.616: I/godot(5539): 	-mx		 Request Maximized
08-30 16:20:37.616: I/godot(5539): 	-w		 Request Windowed
08-30 16:20:37.616: I/godot(5539): 	-vd DRIVER	 : Video Driver (
08-30 16:20:37.616: I/godot(5539): GLES2
08-30 16:20:37.616: I/godot(5539): )
08-30 16:20:37.616: I/godot(5539): 	-ldpi	 : Force low-dpi mode (OSX Only)
08-30 16:20:37.616: I/godot(5539): 	-ad DRIVER	 : Audio Driver (
08-30 16:20:37.616: I/godot(5539): Android
08-30 16:20:37.616: I/godot(5539): )
08-30 16:20:37.616: I/godot(5539): 	-rthread <mode>	 : Render Thread Mode ('unsafe', 'safe', 'separate').
08-30 16:20:37.616: I/godot(5539): 	-s,-script [script] : Run a script.
08-30 16:20:37.616: I/godot(5539): 	-d,-debug : Debug (local stdout debugger).
08-30 16:20:37.616: I/godot(5539): 	-rdebug ADDRESS : Remote debug (<ip>:<port> host address).
08-30 16:20:37.616: I/godot(5539): 	-fdelay [msec]: Simulate high CPU load (delay each frame by [msec]).
08-30 16:20:37.616: I/godot(5539): 	-timescale [msec]: Simulate high CPU load (delay each frame by [msec]).
08-30 16:20:37.616: I/godot(5539): 	-bp : breakpoint list as source::line comma separated pairs, no spaces (%20,%2C,etc instead).
08-30 16:20:37.616: I/godot(5539): 	-v : Verbose stdout mode
08-30 16:20:37.616: I/godot(5539): 	-lang [locale]: Use a specific locale
08-30 16:20:37.616: I/godot(5539): 	-rfs <host/ip>[:<port>] : Remote FileSystem.
08-30 16:20:37.616: I/godot(5539): 	-rfs_pass <password> : Password for Remote FileSystem.
08-30 16:20:37.630: I/godot(5539): [1;31mERROR: free_static: [0m[1mCondition ' !MemoryPoolStatic::get_singleton() ' is true.
08-30 16:20:37.631: I/godot(5539): [0;31m   At: core/os/memory.cpp:59.[0m
08-30 16:20:37.631: I/godot(5539): [1;31mERROR: free_static: [0m[1mCondition ' !MemoryPoolStatic::get_singleton() ' is true.
08-30 16:20:37.631: I/godot(5539): [0;31m   At: core/os/memory.cpp:59.[0m
08-30 16:20:37.631: I/godot(5539): [1;31mERROR: free_static: [0m[1mCondition ' !MemoryPoolStatic::get_singleton() ' is true.
08-30 16:20:37.631: I/godot(5539): [0;31m   At: core/os/memory.cpp:59.[0m
08-30 16:20:37.631: I/godot(5539): *****UNABLE TO SETUP
08-30 16:20:37.650: E/SensorManager(5539): sensor or listener is null
08-30 16:20:37.691: E/SensorManager(5539): sensor or listener is null
08-30 16:20:37.725: D/AndroidRuntime(5539): Shutting down VM
08-30 16:20:37.725: W/dalvikvm(5539): threadid=1: thread exiting with uncaught exception (group=0x419bac98)
08-30 16:20:37.725: W/dalvikvm(5539): threadid=1: uncaught exception occurred
08-30 16:20:37.726: W/System.err(5539): java.lang.RuntimeException: Unable to resume activity {org.asmstudio.spacecute/org.godotengine.godot.Godot}: java.lang.NullPointerException
08-30 16:20:37.733: W/System.err(5539): 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:2964)
08-30 16:20:37.733: W/System.err(5539): 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:2993)
08-30 16:20:37.734: W/System.err(5539): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2395)
08-30 16:20:37.734: W/System.err(5539): 	at android.app.ActivityThread.access$800(ActivityThread.java:151)
08-30 16:20:37.734: W/System.err(5539): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1321)
08-30 16:20:37.734: W/System.err(5539): 	at android.os.Handler.dispatchMessage(Handler.java:110)
08-30 16:20:37.734: W/System.err(5539): 	at android.os.Looper.loop(Looper.java:193)
08-30 16:20:37.734: W/System.err(5539): 	at android.app.ActivityThread.main(ActivityThread.java:5292)
08-30 16:20:37.734: W/System.err(5539): 	at java.lang.reflect.Method.invokeNative(Native Method)
08-30 16:20:37.734: W/System.err(5539): 	at java.lang.reflect.Method.invoke(Method.java:515)
08-30 16:20:37.734: W/System.err(5539): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:824)
08-30 16:20:37.734: W/System.err(5539): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:640)
08-30 16:20:37.734: W/System.err(5539): 	at dalvik.system.NativeStart.main(Native Method)
08-30 16:20:37.735: W/System.err(5539): Caused by: java.lang.NullPointerException
08-30 16:20:37.735: W/System.err(5539): 	at org.godotengine.godot.Godot.onResume(Godot.java:630)
08-30 16:20:37.735: W/System.err(5539): 	at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1197)
08-30 16:20:37.735: W/System.err(5539): 	at com.lbe.security.service.core.client.b.x.callActivityOnResume(Unknown Source)
08-30 16:20:37.735: W/System.err(5539): 	at android.app.Activity.performResume(Activity.java:5343)
08-30 16:20:37.735: W/System.err(5539): 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:2950)
08-30 16:20:37.736: W/System.err(5539): 	... 12 more
08-30 16:20:37.736: W/dalvikvm(5539): threadid=1: calling UncaughtExceptionHandler
08-30 16:20:37.740: E/AndroidRuntime(5539): FATAL EXCEPTION: main
08-30 16:20:37.740: E/AndroidRuntime(5539): Process: org.asmstudio.spacecute, PID: 5539
08-30 16:20:37.740: E/AndroidRuntime(5539): java.lang.RuntimeException: Unable to resume activity {org.asmstudio.spacecute/org.godotengine.godot.Godot}: java.lang.NullPointerException
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:2964)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:2993)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2395)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at android.app.ActivityThread.access$800(ActivityThread.java:151)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1321)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at android.os.Handler.dispatchMessage(Handler.java:110)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at android.os.Looper.loop(Looper.java:193)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at android.app.ActivityThread.main(ActivityThread.java:5292)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at java.lang.reflect.Method.invokeNative(Native Method)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at java.lang.reflect.Method.invoke(Method.java:515)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:824)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:640)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at dalvik.system.NativeStart.main(Native Method)
08-30 16:20:37.740: E/AndroidRuntime(5539): Caused by: java.lang.NullPointerException
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at org.godotengine.godot.Godot.onResume(Godot.java:630)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1197)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at com.lbe.security.service.core.client.b.x.callActivityOnResume(Unknown Source)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at android.app.Activity.performResume(Activity.java:5343)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:2950)
08-30 16:20:37.740: E/AndroidRuntime(5539): 	... 12 more
```

9) SAD

**Comment by RandomShaper** [comment]
Thanks for the report! I'll try to make you happy again as soon as possible. :) Maybe even today.

**Comment by DrMoriarty** [comment]
@RandomShaper 
I just checked this PR (https://github.com/godotengine/godot/pull/10792) and it works for godot test projects. At least I checked monitors, profiler and live scene updating. All of them works as expected.

But it always crashes on my own project:

```
08-30 22:18:30.718: W/dalvikvm(7467): Exception thrown (Ljava/lang/NoClassDefFoundError;) while throwing internal exception (Ljava/lang/OutOfMemoryError;)
08-30 22:18:30.718: I/godot(7467): [1;31mERROR: set_locale: [0m[1mInvalid Locale: 
08-30 22:18:30.718: I/godot(7467): [0;31m   At: core/translation.cpp:921.[0m
08-30 22:18:30.726: D/GraphicBuffer(7467): close handle(0x602b1668) (w:720 h:1280 f:1)
08-30 22:18:30.732: D/dalvikvm(7467): threadid=11: exiting
08-30 22:18:30.732: W/dalvikvm(7467): threadid=11: thread exiting with uncaught exception (group=0x419bac98)
08-30 22:18:30.732: W/dalvikvm(7467): threadid=11: uncaught exception occurred
08-30 22:18:30.733: W/System.err(7467): java.lang.NoClassDefFoundError: java/util/Locale
08-30 22:18:30.736: W/System.err(7467): 	at org.godotengine.godot.GodotIO.getLocale(GodotIO.java:516)
08-30 22:18:30.736: W/System.err(7467): 	at org.godotengine.godot.GodotLib.step(Native Method)
08-30 22:18:30.736: W/System.err(7467): 	at org.godotengine.godot.GodotView$Renderer.onDrawFrame(GodotView.java:618)
08-30 22:18:30.736: W/System.err(7467): 	at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1557)
08-30 22:18:30.736: W/System.err(7467): 	at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1263)
08-30 22:18:30.736: W/System.err(7467): Caused by: java.lang.ClassNotFoundException: Didn't find class "org/godotengine/godot/GodotAppodeal" on path: DexPathList[[zip file "/data/app/org.asmstudio.spacecute-1.apk"],nativeLibraryDirectories=[/data/app-lib/org.asmstudio.spacecute-1, /vendor/lib, /system/lib]]
08-30 22:18:30.739: W/System.err(7467): 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
08-30 22:18:30.739: W/System.err(7467): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:497)
08-30 22:18:30.739: W/System.err(7467): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:457)
08-30 22:18:30.740: W/System.err(7467): 	... 4 more
08-30 22:18:30.740: W/dalvikvm(7467): threadid=11: calling UncaughtExceptionHandler
08-30 22:18:30.742: E/AndroidRuntime(7467): FATAL EXCEPTION: GLThread 36633
08-30 22:18:30.742: E/AndroidRuntime(7467): Process: org.asmstudio.spacecute, PID: 7467
08-30 22:18:30.742: E/AndroidRuntime(7467): java.lang.NoClassDefFoundError: java/util/Locale
08-30 22:18:30.742: E/AndroidRuntime(7467): 	at org.godotengine.godot.GodotIO.getLocale(GodotIO.java:516)
08-30 22:18:30.742: E/AndroidRuntime(7467): 	at org.godotengine.godot.GodotLib.step(Native Method)
08-30 22:18:30.742: E/AndroidRuntime(7467): 	at org.godotengine.godot.GodotView$Renderer.onDrawFrame(GodotView.java:618)
08-30 22:18:30.742: E/AndroidRuntime(7467): 	at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1557)
08-30 22:18:30.742: E/AndroidRuntime(7467): 	at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1263)
08-30 22:18:30.742: E/AndroidRuntime(7467): Caused by: java.lang.ClassNotFoundException: Didn't find class "org/godotengine/godot/GodotAppodeal" on path: DexPathList[[zip file "/data/app/org.asmstudio.spacecute-1.apk"],nativeLibraryDirectories=[/data/app-lib/org.asmstudio.spacecute-1, /vendor/lib, /system/lib]]
08-30 22:18:30.742: E/AndroidRuntime(7467): 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
08-30 22:18:30.742: E/AndroidRuntime(7467): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:497)
08-30 22:18:30.742: E/AndroidRuntime(7467): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:457)
08-30 22:18:30.742: E/AndroidRuntime(7467): 	... 4 more
```

**Comment by DrMoriarty** [comment]
@RandomShaper sorry, it was my mistake. Last time I built vanilla godot without all my modules.

I merged your PR to my branch so now it worked like a charm!

Thank you very much!

**Comment by RandomShaper** [comment]
Thanks to you for the updates!

**Comment by eon-s** [comment]
Amazing @RandomShaper !

A few things to add:
- Works with usb, just need the ip where adb is running in network like if it were a wifi
- The small deploy never worked for me before this! :feelsgood: 
- Older settings may keep the deprecated entry on `Editor settings>Android` visible.
- We won't be able to test if this work with Godot 3 until 3.1 with our devices, just a few below 5.0 support GLES3.
- Remember @akien-mga to merge the PR after he returns from the post release slumber :P

**Comment by RandomShaper** [comment]
Could you elaborate the first point? I'm not sure I get what you mean.

**Comment by eon-s** [comment]
@RandomShaper is supposed to work over wifi only on versions < 5.0, right?

**Comment by RandomShaper** [comment]
Yes:

- < 5, Wi-Fi (standard networking, even Ethernet if the device featured it)
- >= 5, USB (through `adb reverse`)

**Comment by eon-s** [comment]
Ok, well, < 5 works with USB too, only needs to set up the pc IP on "network" (no idea why it works).

**Comment by RandomShaper** [comment]
Maybe it works because the device has USB tethering for Internet enabled.

**Comment by RandomShaper** [comment]
I wondered if assuming networked debugging to be over Wi-Fi was to inaccurate, but I guess it's more understandable for the common user. I mean, what the message says.

**Comment by eon-s** [comment]
Instead of "wi-fi" can say "network", it may be a hint that needs to set up network correctly.

**Comment by RandomShaper** [comment]
I still think that may be too abstract for the not-very-techie user. I'd like to hear more opinions and rather let it be in the meantime.

**Comment by RandomShaper** [comment]
Since the main issue is now solved by #10792 I'm closing this.

In spite of that, we can still discuss here about the wording.
